Source/AtomicEditor/Editors/SceneEditor3D/SceneEditor3D.cpp

```diff
@@ -251,8 +255,28 @@ bool SceneEditor3D::Save()
         file.Close();
     }
 
+    SetModified(false);
+
     return true;
 
 }
 
+void SceneEditor3D::SetModified(bool modified)
+{
+    ResourceEditor::SetModified(modified);
+
+    if (modified)
+    {
+        String filename = GetFileNameAndExtension(fullpath_);
+        filename += "*";
+        button_->SetText(filename.CString());
+    }
+    else
+    {
+        String filename = GetFileNameAndExtension(fullpath_);
+        button_->SetText(filename.CString());
+    }
+
+}
```

Comment:
We can probably hook into the undo/redo system to mark a scene as modified (on any undo/redo data creation, mark the editor as modified)